### ISSUE TITLE
hydra: ssh is used as a launcher on a single node 

### ISSUE THREAD

**Issue by orcunyildiz** [pmodels/mpich#7186]
I'm running mpich on a single node on the LCRC Swing cluster, but the launcher still picks up the ssh instead of fork. Please ignore the OpenSSL error, which is due to mismatch in OpenSSL versions between my Spack environment and Swing.

```
oyildiz@gpu6: $ mpiexec -n 12 ./lmp -in ./in.relax
OpenSSL version mismatch. Built against 30000020, you have 30100030
[mpiexec@gpu6] ui_cmd_cb (mpiexec/pmiserv_pmci.c:51): Launch proxy failed.
[mpiexec@gpu6] HYDT_dmxu_poll_wait_for_event (lib/tools/demux/demux_poll.c:76): callback returned error status
[mpiexec@gpu6] HYD_pmci_wait_for_completion (mpiexec/pmiserv_pmci.c:181): error waiting for event
[mpiexec@gpu6] main (mpiexec/mpiexec.c:252): process manager error waiting for completion
```
Here is the launcher details from the verbose mode output: 

```
[mpiexec@gpu6] Launch arguments: /bin/ssh -x gpu6.lcrc.anl.gov "/lcrc/project/PEDAL/oyildiz/software/spack/var/spack/environments/accelerate-spack-env-mpi-cuda/.spack-env/view/bin/hydra_pmi_proxy" --control-port gpu6:44647 --debug --rmk pbs --launcher ssh --demux poll --pgid 0 --retries 10 --usize -2 --pmi-port 0 --gpus-per-proc -2 --gpu-subdevs-per-proc -2 --proxy-id 0 
```

**Comment by raffenet** [comment]
@orcunyildiz I lost track of this but do want to note you can explicitly specify the fork launcher by with `mpiexec -launcher fork`.

**Comment by orcunyildiz** [comment]
Thanks, Ken. I think our discussion was if the mpiexec is called on a single node it should pick up the `fork` launcher instead of ssh.

**Comment by raffenet** [comment]
> Thanks, Ken. I think our discussion was if the mpiexec is called on a single node it should pick up the `fork` launcher instead of ssh.

Agreed. It is not a trivial change since we set internally set the launcher first before detecting the node list. I need to do some testing to see if resetting it after determining a single node environment will work.

**Comment by raffenet** [comment]
@orcunyildiz I created https://github.com/pmodels/mpich/pull/7513 to try and address this. After working with a student on Swing, we identified another issue when trying to use the native PBS launcher mechanism, but that should only matter if you are using multiple nodes. We'll fix that in another PR and also will add a using MPICH on Swing wiki page for users.

**Comment by orcunyildiz** [comment]
Great! Thanks again, Ken.
